cypress/e2e/context-navigation.cy.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+let localUser
+let nonLocalUser
+let contextTitlePrefix = 'test application'
+let contextTitle = contextTitlePrefix
+let testNumber = 0
+
+describe('Test context navigation', () => {
+    before(function () {
+        cy.createRandomUser().then(user => {
+            localUser = user
+        })
+
+        cy.createRandomUser().then(user => {
+            nonLocalUser = user
+        })
+    })
+
+    beforeEach(function () {
+        testNumber += 1
+        contextTitle = contextTitlePrefix + ' ' + testNumber
+        cy.login(localUser)
+        cy.visit('apps/tables')
+        cy.wait(1000)
```

Comment:
could this cause flakyness? is there a way to wait that `visit ()` was completed and the page loaded ready?

Reply:
Yes, that is an anti-pattern, should rather wait for a specific element to appear for example.